rust/protocol/src/address.rs

```diff
@@ -1,26 +1,57 @@
 //
-// Copyright 2020 Signal Messenger, LLC.
+// Copyright 2020-2021 Signal Messenger, LLC.
 // SPDX-License-Identifier: AGPL-3.0-only
 //
 
+//! Where to send a [sealed_sender] message to.
+
+// These are used in docstrings.
+#[allow(unused_imports)]
+use crate::sealed_sender::{self, UnidentifiedSenderMessageContent};
+
 use std::fmt;
 
+/// The type used in memory to represent a *device*, i.e. a particular Signal client instance which
+/// represents some user.
+///
+/// Used in [ProtocolAddress].
+pub type DeviceId = u32;
```

Comment:
Thank you for this; it definitely needs to propagate further!

Reply:
Normally when I write Rust code like this, I actually avoid type aliases and instead use wrapper structs and `From` impls. Everything can still be `Copy`, but I think you get a form of type safety when you do this, e.g.:
```rust
#[derive(Copy, Clone)]
pub struct DeviceId(u32);
impl From<u32> for DeviceId { ... }
impl From<DeviceId> for u32 { ... }
```

Would you accept that sort of change in a followup PR (to focus this one on just doc fixes)? Normally I would expect to see a type alias used as here when using `#[repr(C)]` or other things where you want a very clear struct layout. In the implementation of this API (essentially everywhere except the `rust/bridge/` crate), I think we can make the effort to use explicit wrapper structs with explicit `From` impls.

However, if that sort of larger change is not a priority right now, this type alias is a good intermediate solution I think.

Reply:
I was thinking about that too. It can't propagate all the way into all three app languages because Java and TypeScript don't have zero-cost wrapper types, but that's no reason not to do it in Rust, so sure, I'd take a follow-up for that.

Reply:
Awesome, thanks!!

Reply:
Broke this out into #289!